test/Bitmaps/BitmapFile.test.cpp

```diff
@@ -1,17 +1,60 @@
 #include "../src/Bitmaps/BitmapFile.h"
+#include "XFile.h"
 #include <gtest/gtest.h>
 #include <stdexcept>
 
+void RoundTripSub(uint16_t bitCount, int32_t width, int32_t height)
+{
+	const std::string filename("Sprites/data/BitmapTest.bmp");
+
+	BitmapFile bitmapFile = BitmapFile::CreateDefaultIndexed(bitCount, width, height);
+	BitmapFile bitmapFile2;
+
+	EXPECT_NO_THROW(BitmapFile::WriteIndexed(filename, bitmapFile));
+	EXPECT_NO_THROW(bitmapFile2 = BitmapFile::ReadIndexed(filename));
+	EXPECT_EQ(bitmapFile, bitmapFile2);
+
+	XFile::DeletePath(filename);
+}
+
+TEST(BitmapFile, RoundTripWriteAndRead)
+{
+	/* Test cases:
+		 * Width below pitch
+		 * Greater than 1 height
+		 * Width equal to pitch
+		 * Width above pitch
+	*/
+	{
+		SCOPED_TRACE("Monochrome");
+		RoundTripSub(1, 1, 1);
+		RoundTripSub(1, 1, 2);
+		RoundTripSub(1, 32, 1);
+		RoundTripSub(1, 33, 1);
+	}
+	{
+		SCOPED_TRACE("4 Bit");
+		RoundTripSub(4, 1, 1);
+		RoundTripSub(4, 1, 2);
+		RoundTripSub(4, 8, 1);
+		RoundTripSub(4, 9, 1);
+	}
+	{
+		SCOPED_TRACE("8 Bit");
+		RoundTripSub(8, 1, 1);
+		RoundTripSub(8, 1, 2);
+		RoundTripSub(8, 2, 1);
+	}
```

Comment:
It's actually pretty neat how you did this. Interesting use of Google Test features.